sidechain/test/src/sidechain_block_builder.rs

```diff
@@ -111,13 +113,27 @@ impl SidechainBlockBuilder {
 		self
 	}
 
+	/// Calculate the payload of a sidechain block.
+	pub fn block_data_hash(&self) -> H256 {
+		(
+			self.timestamp,
+			self.parentchain_block_hash,
+			self.signer.public(),
+			self.signed_top_hashes.as_slice(),
+			self.encrypted_payload.as_slice(),
+		)
+			.using_encoded(BlakeTwo256::hash)
+	}
+
 	pub fn build(&self) -> Block {
+		let block_data_hash = self.block_data_hash();
+
+		let header = HeaderTrait::new(self.number, self.parent_hash, self.shard, block_data_hash);
```

Comment:
This is not the way a builder pattern usually works, but for time's sake we'll leave it as is for the moment